model/artifact/artifact_file.go

```diff
@@ -96,10 +87,26 @@ func StripHiddenFiles(files []File, hasUser bool) ([]File, error) {
 	return publicFiles, nil
 }
 
-// ContainsSigningParams returns true if all the params needed for
-// presigning a url are present
-func (f *File) ContainsSigningParams() bool {
-	return !(f.AwsSecret == "" || f.AwsKey == "" || f.Bucket == "" || f.FileKey == "")
+func presignFile(file File) (string, error) {
+	if file.AwsSecret == "" || file.AwsKey == "" || file.Bucket == "" || file.FileKey == "" {
+		return "", errors.New("AWS secret, AWS key, S3 bucket, or file key missing")
+	}
+
+	// TODO (DEVPROD-5758): remove this special casing once artifacts from the old
```

Comment:
I'll open a different ticket for this TODO once we come to consensus that this is a good idea.

Reply:
Done: [DEVPROD-6193](https://jira.mongodb.org/browse/DEVPROD-6193)